library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -80,21 +81,26 @@ const MAXIMUM_ZST_CAPACITY: usize = 1 << (usize::BITS - 1); // Largest possible
 /// [`make_contiguous`]: VecDeque::make_contiguous
 #[cfg_attr(not(test), rustc_diagnostic_item = "vecdeque_type")]
 #[stable(feature = "rust1", since = "1.0.0")]
-pub struct VecDeque<T> {
+pub struct VecDeque<
+    T,
+    #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global,
```

Comment:
You could avoid quite a lot of boilerplate by only adding the : Allocator constraint in the functions you use it, instead of when you define the struct.

Reply:
No, because `A: Allocator` bound is required by `Drop`, so it needs to be on the struct definition